co-authors-plus.php

```diff
@@ -766,7 +766,10 @@ function coauthors_set_post_author_field( $data, $postarr ) {
 
 		// This action happens when a post is saved while editing a post
 		if ( isset( $_REQUEST['coauthors-nonce'] ) && isset( $_POST['coauthors'] ) && is_array( $_POST['coauthors'] ) ) {
-			$author = sanitize_text_field( $_POST['coauthors'][0] );
+
+			// urlencode() is for encoding coauthor name with special characters to compare names when getting coauthor.
+			$author = urlencode( sanitize_text_field( $_POST['coauthors'][0] ) );
```

Comment:
Only legacy systems should use `urlencode()`. Instead, `rawurlencode()` should be used whenever a URL needs to be properly encoded.

Reply:
Replaced `urlencode/urldecode` with `rawurlencode/rawurldecode` in 9806b14f6e2169a50b43082e73aa9445f2ce1ea3